1.2.0 pre-release changes (#365)
signalfx/splunk-otel-java: CHANGELOG.md

## Unreleased

## v1.2.0 - 2021-07-26

### General

> - OpenTelemetry Java SDK and OpenTelemetry Instrumentation for Java dependencies were updated to version 1.4.0.

Reviewer: ```suggestion
- OpenTelemetry Java SDK and OpenTelemetry Instrumentation for Java dependencies have been updated to version 1.4.0.
```

### Enhancements

- We have added instrumentations for [HikariCP](https://github.com/brettwooldridge/HikariCP)
  and [Tomcat JDBC](https://tomcat.apache.org/tomcat-8.5-doc/jdbc-pool.html) connection pools. The agent will now
  collect and export metrics for those two JDBC connection pools.
- The agent now supports the `OTEL_SERVICE_NAME` environment variable (and `otel.service.name` system property) - you no

>   longer need to use `OTEL_RESOURCE_ATTRIBUTES` to set the service name.

Reviewer 2: should we mention that this new way is a preferred one?

Reviewer: Discussed in Slack.

> ### Enhancements
> 
> - We have added instrumentations for [HikariCP](https://github.com/brettwooldridge/HikariCP)
>   and [Tomcat JDBC](https://tomcat.apache.org/tomcat-8.5-doc/jdbc-pool.html) connection pools. The agent will now
>   collect and export metrics for those two JDBC connection pools.
> - The agent now supports the `OTEL_SERVICE_NAME` environment variable (and `otel.service.name` system property) - you no
>   longer need to use `OTEL_RESOURCE_ATTRIBUTES` to set the service name.

Reviewer: ```suggestion
### Enhancements

- We have added instrumentation for [HikariCP](https://github.com/brettwooldridge/HikariCP)
  and [Tomcat JDBC](https://tomcat.apache.org/tomcat-8.5-doc/jdbc-pool.html) connection pools. The agent now
  collects and exports metrics for both JDBC connection pools.
- You can now set the service name using the `OTEL_SERVICE_NAME` environment variable and the `otel.service.name` system property. This removes the need of using `OTEL_RESOURCE_ATTRIBUTES` to set the service name.
```
